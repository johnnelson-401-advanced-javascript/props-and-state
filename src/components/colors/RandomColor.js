import React, { Component } from 'react';

export default class RandomColor extends Component {
  state = {
    backgroundColor: 'red'
  };

  randomColor = () =>{
    var color = '#';
    const letters = '0123456789ABCDEF';
    for(let i = 0; i < 6; i++) {
      color += letters[Math.floor(Math.random() * 16)];
    }
    this.setState({ backgroundColor: color });
  } 

  componentDidMount() {
    setInterval(function() {
    }, 1000);
  }

  render() {
    return (
      <>
        <div style={{
          backgroundColor: this.state,
          width: '100px',
          height: '100px'
        }}></div>
      </>
    );
  }
}
