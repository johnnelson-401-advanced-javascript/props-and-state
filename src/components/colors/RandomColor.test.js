import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('RandomColor', () => {
  it('Renders a random Color', () => {
    let wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
