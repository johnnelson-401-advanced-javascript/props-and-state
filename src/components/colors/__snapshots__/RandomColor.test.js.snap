// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor Renders a random Color 1`] = `
<Fragment>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "height": "100px",
        "width": "100px",
      }
    }
  />
</Fragment>
`;
